docs/admin/node-allocatable.md

```diff
@@ -0,0 +1,144 @@
+
+### Kube Reserved
+
+**Kubelet Flag**: `--kube-reserved=[cpu=100mi][,][memory=100Mi]`
+**Kubelet Flag**: `--kube-reserved-cgroup=`/runtime.slice`
+
+`kube-reserved` is meant to capture resource reservation for kubernetes system daemons like the `kubelet`, `container runtime`, `node problem detector`, etc.
+It is not meant to reserve resources for system daemons that are run as pods.
+`kube-reserved` is typically a function of `pod density` on the nodes.
+[This performance dashboard](http://node-perf-dash.k8s.io/#/builds) exposes `cpu` and `memory` usage profiles of `kubelet` and `docker engine` at multiple levels of pod density.
+[This blog post](http://blog.kubernetes.io/2016/11/visualize-kubelet-performance-with-node-dashboard.html) explains how the dashboard can be interpreted to come up with a suitable `kube-reserved` reservation.
+
+It is recommended that the kubernetes system daemons are placed under a top level control group (`system.slice` on systemd machines for example).
+Each system daemon should ideally run within its own child control group.
+Refer to [this doc](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/node-allocatable.md#recommended-cgroups-setup) for more details on recommended control group hierarchy.
+
+To optionally enforce `kube-reserved` on system daemons, specify the parent control group for kube daemons as the value for `--kube-reserved-cgroup` kubelet flag.
+
+### System Reserved
+
+**Kubelet Flag**: `--system-reserved=[cpu=100mi][,][memory=100Mi]`
+**Kubelet Flag**: `--system-reserved-cgroup=`/system.slice`
+
+
+`system-reserved` is meant to capture resource reservation for OS system daemons like `sshd`, `udev`, etc.
+`system-reserved` should reserve `memory` for the `kernel` too since `kernel` memory is not accounted to pods (yet) in Kubernetes.
+Reserving resources for user login sessions is also recommended (`user.slice` in systemd world).
+
+To optionally enforce `system-reserved` on system daemons, specify the parent control group for OS system daemons as the value for `--system-reserved-cgroup` kubelet flag.
+
+### Eviction Thresholds
+
+**Kubelet Flag**: `--eviction-hard=[memory.available<500Mi]`
+
+Memory pressure at the node level leads to System OOMs which affects the entire node and all pods running on it.
+Nodes can go offline temporarily until memory has been reclaimed.
+To avoid (or reduce the probabilty) system OOMs kubelet provides [`Out of Resource`](./out-of-resource.md) management.
+Evictions are supported for `memory` and `storage` only.
+By reserving some memory via `--eviction-hard` flag, the `kubelet` attempts to `evict` pods whenever memory availability on the node drops below the reserved value.
+Hypothetically, if system daemons did not exist on a node, pods cannot use more than `capacity - eviction-hard`.
+For this reason, resources reserved for evictions will not be available for pods.
```

Comment:
Scheduling is meant to be implicit since pods can be placed directly on nodes bypassing the scheduler